lib/Connection.php

```diff
@@ -335,6 +344,18 @@ public function query($sql, &$values=array())
 				throw new DatabaseException($this);
 		} catch (PDOException $e) {
 			throw new DatabaseException($e);
+		} finally {
```

Comment:
I think that's a decent idea. But at the same time I'd probably just make it a private method at that point.